test/services/katello/pulp3/repository_integration_test.rb

```diff
@@ -0,0 +1,59 @@
+require 'katello_test_helper'
+require 'support/pulp3_support'
+require 'katello/pulp3/repository'
+
+types = Katello::RepositoryTypeManager.defined_repository_types
+types.slice!(ENV['TEST_CONTENT_TYPES'].split(',')) unless ENV['CONTENT_TYPES'].blank?
+types.values.each do |repository_type|
+  if repository_type.test_url
+    class_name = "#{repository_type.id.to_s.camelize}TypeSyncTest"
+    ::Katello::Pulp3::Repository.const_set(class_name, Class.new(ActiveSupport::TestCase) do
+      include Katello::Pulp3Support
+
+      #REPO_TYPE constant is set after class is defined
+      def repo_type
+        self.class::REPO_TYPE
+      end
+
+      def setup
+        User.current = users(:admin)
+        @primary = SmartProxy.pulp_primary
+        @repo = katello_repositories(:fedora_17_x86_64_duplicate)
+        @repo.root.update!(url: repo_type.test_url, :content_type => repo_type.id,
+                           :download_policy => nil, generic_remote_options: {})
+        @repo.root.update(repo_type.test_url_root_options) if repo_type.test_url_root_options
+        @repo.update(:pulp_id => @repo.pulp_id + "-test-#{repo_type.id}", :relative_path => "/integration_tests/#{repo_type.id}")
+        create_repo(@repo, @primary)
+        ForemanTasks.sync_task(::Actions::Katello::Repository::MetadataGenerate, @repo)
+      end
+
+      def teardown
+        User.as_anonymous_admin do
+          ForemanTasks.sync_task(
+            ::Actions::Pulp3::Orchestration::Repository::Delete, @repo, @primary)
+        end
+      end
+
+      def test_sync
+        @repo.update(publication_href: nil, version_href: nil)
+        ForemanTasks.sync_task(::Actions::Katello::Repository::Sync, @repo)
+        @repo.reload
+        refute_equal @repo.version_href, @repo_version_href
```

Comment:
does `@repo_version_href` exist?

Reply:
i don't think !  this line isn't really doing much, i'll remove it

Reply:
good eye!